modules/objdetect/test/test_qrcode_encode.cpp

```diff
@@ -430,4 +430,80 @@ TEST(Objdetect_QRCode_Encode_Decode_Structured_Append, DISABLED_regression)
 
 #endif // UPDATE_QRCODE_TEST_DATA
 
+TEST(Objdetect_QRCode_Encode_Decode, regression_issue22029)
+{
+    const cv::String expect_msg = "OpenCV";
+    const int min_version = 1;
+    const int max_version = 40;
+
+    for ( int v = min_version ; v <= max_version ; v++ )
+    {
+        Mat qrimg;
```

Comment:
Please add GoogleTest's scope identifier in "for" loop body (used for error messages):

```
SCOPED_TRACE(cv::format("version=%d", v));
```